src/eventbuffer/InMemoryJSONEventBuffer.ts

```diff
@@ -488,46 +432,75 @@ export default class InMemoryJSONEventBuffer implements EventBuffer<EventData>,
           this.logger.debug(
             `Event Reporting - InMemoryJSONEventBuffer - sendEventImmediately - beaconing data out ${body}`
           );
-          /* istanbul ignore else */
           if (!navigator.sendBeacon(`${this.ingestionURL}?beacon=1`, body)) {
             failed = true;
+          } else {
+            failed = false;
           }
         }
       } catch (error) {
-        this.logger.warn(`Event Reporting - Error sending beacon for an important event ${name}`);
+        this.logger.warn(`Event Reporting - Error sending beacon for an important event ${body}`);
         failed = true;
       }
     }
 
     /* istanbul ignore else */
     if (failed) {
       this.logger.debug(
-        `Event Reporting - InMemoryJSONEventBuffer - sendEventImmediately - pushing to failed events ${JSON.stringify(
-          event
-        )}`
+        `Event Reporting - InMemoryJSONEventBuffer - sendEventImmediately - pushing to failed events ${body}`
       );
       this.failedIngestionEvents.push(event);
     }
   }
 
+  private getBackoffWaitTime(retryCount: number): number {
+    return (
+      Math.random() *
+      Math.min(
+        InMemoryJSONEventBuffer.RETRY_LONG_BACKOFF_MS,
+        InMemoryJSONEventBuffer.RETRY_SHORT_BACKOFF_MS * Math.pow(2.0, retryCount)
+      )
+    );
+  }
+
+  private wait(delayInMs: number): Promise<void> {
```

Comment:
nit: Should we move this to an util folder?

Reply:
Actually I felt that the timeoutScheduler implementation here is a little complex than needed. So, based on the timeout I have pass a callback function when I do `timeoutScheduler.start(() => { call send again here })` hence, used this. I preferred not to separate this function since we do not have any other util folder. Could you please let me know what should I do here then?

Reply:
As discussed have added a utils folder with this function. But we do not export it publicly.